tests/dbus/test-client

```diff
@@ -79,6 +79,14 @@ update_meshcop_txt_and_check()
     service="$(scan_meshcop_service)"
     grep --binary-files=text "A=abc" <<<"${service}"
     grep --binary-files=text "B=12" <<<"${service}"
+
+    sudo gdbus call --system --dest io.openthread.BorderRouter.wpan0 --method=io.openthread.BorderRouter.Reset --object-path /io/openthread/BorderRouter/wpan0
+    sleep 30
+
+    sudo gdbus call --system --dest io.openthread.BorderRouter.wpan0 --method=io.openthread.BorderRouter.UpdateVendorMeshCopTxtEntries --object-path /io/openthread/BorderRouter/wpan0 "[('C',[97,98,99])]"
+    sleep 5
+    service="$(scan_meshcop_service)"
+    grep --binary-files=text "C=abc" <<<"${service}"
```

Comment:
Shall we also add tests for discovery proxy?

Reply:
Currently this fix is kind of flaky with mDNSResponder. I'll add the test cases later in another PR.